src/cockpit/samples.py

```diff
@@ -121,21 +126,24 @@ def detect_cpu_sensors(self, hwmonid: int, name: str) -> None:
                 if name != 'cpu_thermal':
                     continue
 
-            self.sensors.append(sensor_path)
+            self.sensors.append(entry.path)
 
-    def sample(self, samples: Samples) -> None:
+    def sample(self, samples: Samples, hwmon_path='/sys/class/hwmon') -> None:
         cpu_names = ['coretemp', 'cpu_thermal', 'k8temp', 'k10temp', 'atk0110']
 
         if not self.sensors:
-            # TODO: 2 ** 32?
-            for index in range(0, 2 ** 32):
+            for path in os.listdir(hwmon_path):
```

Comment:
I'm gonna have to insist on some `*at()` variant stuff here, please.

Reply:
I mean please use more dirfd and less path pasting.

Reply:
Hmmmm `os.fwalk'?

```
>>> list(os.fwalk('/sys/class/hwmon'))
[('/sys/class/hwmon', ['hwmon8', 'hwmon6', 'hwmon10', 'hwmon4', 'hwmon2', 'hwmon0', 'hwmon9', 'hwmon7', 'hwmon5', 'hwmon3', 'hwmon1'], [], 3)]
```

doesn't give me much dirfd improvements tbh. The dirfd is from "/sys/class/hwmon" while I think you want it from "/sys/class/hwmon1".

```
>>> list(os.fwalk('/sys/class/hwmon/hwmon3/'))
[('/sys/class/hwmon/hwmon3/', ['power', 'device', 'subsystem'], ['uevent', 'temp2_min', 'temp2_max', 'temp3_input', 'temp3_min', 'temp1_label', 'temp1_crit', 'temp3_max', 'temp1_alarm', 'temp2_label', 'temp1_input', 'temp1_min', 'temp3_label', 'temp1_max', 'name', 'temp2_input'], 3), ('/sys/class/hwmon/hwmon3/power', [], ['runtime_active_time', 'runtime_status', 'autosuspend_delay_ms', 'runtime_suspended_time', 'control'], 4)]
```

And then loop over files (ignore the second loop as it's `hwmon3/power/`, I don't think fwalk can limit it's iteration.